pkg/plugin/aggregation/run_test.go

```diff
@@ -197,3 +198,87 @@ func getTestCert() (*tls.Certificate, error) {
 		PrivateKey:  privKey,
 	}, nil
 }
+
+type MockCleanupPlugin struct {
```

Comment:
Why create a whole new plugin struct here instead of just referencing jobs or daemonset plugin types?

Reply:
I started out using a job plugin type but realised that taking that approach would require this test to have knowledge of the internals of the job's `Cleanup` method in order to check that it was called (by checking that the [`CleanedUp` field](https://github.com/heptio/sonobuoy/blob/master/pkg/plugin/driver/job/job.go#L179) was set). For this test, I only wanted to check that the call was skipped correctly. Given that the `Cleanup` function takes a slice of `plugin.Interface`, it seemed less brittle to provide a mock implementation of that interface rather than rely on the implementation of `job.Cleanup`.

Reply:
> knowledge of the internals of the job's Cleanup method

You're referring to the `cleanup` field?

That is somewhat minor IMO but since it isn't part of the interface that seems fair enough. Not a blocker on my end and it does end up being a very isolated test, so that's good.